src/rust/engine/workunit_store/src/lib.rs

```diff
@@ -100,3 +104,18 @@ pub fn get_parent_id() -> Option<String> {
     (*task_parent_id).clone()
   })
 }
+
+#[cfg(test)]
+mod tests {
+  use crate::hex_16_digit_string;
+
+  #[test]
+  fn workunit_span_id_has_16_digits_len_hex_format() {
+    let number: u64 = 1;
+    let hex_string = hex_16_digit_string(number);
+    assert_eq!(16, hex_string.len());
+    for ch in hex_string.chars() {
+      assert!(ch.is_ascii_hexdigit())
```

Comment:
This commit is looking good to me.
Maybe the test could be a tad more stringent, for instance by verifying that the hex encoding turned out OK.
You could to this by taking a somewhat random u64, but writing it down in hex with the 0x prefix, so it's easy to visually check the correctness.
So something like this:
```
assert_eq("0123456789abcdef", hex_16_digit_string(0x0123456789abcdef));
```

Reply:
Since things were green, I went ahead and merged. Can discuss expanding this in a followup. Thanks!